src/GitLabHealth-Model-Importer/GitlabModelImporter.class.st

```diff
@@ -1,1569 +1,1642 @@
+							 
+			              (searchResult class = Dictionary and: [
+				               (searchResult at: #message) includesSubstring:
+					               '403 Forbidden' ])
+				              ifTrue: [ "if the result is an 403 error we fake a new user"
+					              self glhModel
+						              add: (GLHUser new
+								               username: anUsername;
+								               name: anUsername;
+								               yourself)
+						              unless: [ :nu :ou | nu username = ou username ] ]
+				              ifFalse: [
+					              searchResult
+						              ifEmpty: [ "results can be empty thus we force a new user with the info we have "
+							              self glhModel
+								              add: (GLHUser new
+										               username: anUsername;
+										               name: anUsername;
+										               yourself)
+								              unless: [ :nu :ou | nu username = ou username ] ]
+						              ifNotEmpty: [ "because we may already have the researched user, we look by ID in the model"
+							              userId := searchResult first at: #id.
+							              (self glhModel allWithType: GLHUser)
+								              detect: [ :user | user id = userId ]
+								              ifNone: [ self importUser: userId ] ] ] ].
+
+	self userCatalogue addUser: resultUser withName: anUsername.
+
+	^ resultUser
+]
+
+{ #category : #'import-gitblame' }
+GitlabModelImporter >> importUserContributionFromBlames: blamesArray [
+    | dictionary_Emails_Lines line blames   |
+     "filter if  the blame isn't nil  bacause some file like media and photo dosn'thave blame  "
+    blames := blamesArray reject: [:item | item isNil].
+    "Step 1: Initialize dictionary to store email -> total lines count"
+    dictionary_Emails_Lines := Dictionary new.
+    "calculer le nombre totale des ligns "
+    line :=  blames collect: [ :bl |
+    bl  collect: [ :element |
+	    |  lines |
+        lines :=  ((element at: 'lines')size) .
+    ]
+   ].
+    line:=( line flattened) sum .
+
+    "Step 2: Process each blame dictionary"
+     blames do: [:blameEntries |
+        blameEntries isArray ifTrue: [  "Ensure it's a valid blame entry"
+            blameEntries do: [:element | 
+                | authorEmail linesSize |
+                authorEmail := (element at: 'commit') at: 'author_email'.
+                linesSize := (element at: 'lines') size.  "Number of lines modified"
+
+                "Update dictionary (add lines if email exists, otherwise initialize)"
+                (dictionary_Emails_Lines includesKey: authorEmail)
+                    ifTrue: [ dictionary_Emails_Lines at: authorEmail put: ((dictionary_Emails_Lines at: authorEmail) + ((linesSize)/line)asFloat) ]
+                    ifFalse: [ dictionary_Emails_Lines at: authorEmail put:((linesSize)/line)asFloat ].
+            ].
+        ].
+    ].

```

Comment:
This method is outside the scope of the importer. The importer goal is to load object inside the model. Here there is no object added to the model, and no return. The purpose seems to be a boundery effect on the blamesArray only, which is not a collection of the model entities.   
Therefore importUserContributionFromBlames: is a computing methods that should be in another package.